### PR TITLE
Add basic MediaUploadCoordinator

### LINKED ISSUE #7597: Async: Implement Media Upload Coordinator
The Media Upload Coordinator component will encapsulate our existing media creation and upload behaviour, and perform a lot of the logic that currently takes place in the editor and media library view controllers.

This first issue is to create an upload coordinator, wrapping export and upload in a single `addMedia` method.

### PR DESCRIPTION
**Pull request by frosty**
Fixes #7597 

This PR adds a skeleton implementation of a MediaUploadCoordinator. It supports creating and uploading media via a single method, and observing begin / end events for a single Media object or for all Media objects. We'll build on top of this as we go, adding additional functionality and tweaking the implementation.

To test:

* You can check out a branch that makes use of this in the Media Library: `feature/upload-coordinator-in-media-library`. You should be able to use it to add media to the media library, and see a message logged out when the media item is created and finishes uploading.

Needs review: @SergioEstevao

### CODE REVIEW

File: WordPress/Classes/Services/MediaUploadCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    ///
+    /// - parameter onUpdate: A block that will be called whenever media items
+    ///                       (or a specific media item) are updated.
+    /// - parameter media: An optional specific media item to receive updates for.
+    ///                    If provided, the `onUpdate` block will only be called
+    ///                    for updates to this media item, otherwise it will be
+    ///                    called when changes occur to _any_ media item.
+    /// - returns: A UUID that can be used to unregister the observer block at a later time.
+    ///
+    func addObserver(_ onUpdate: @escaping ObserverBlock, for media: Media? = nil) -> UUID {
```
Comment (SergioEstevao) [inline WordPress/Classes/Services/MediaUploadCoordinator.swift]: We probably should refer on the docs here, or in the class docs in what thread the observer blocks will be invoked.

Comment (frosty) [reply WordPress/Classes/Services/MediaUploadCoordinator.swift]: Thanks!

File: WordPress/Classes/Services/MediaUploadCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import Foundation
+
+/// MediaUploadCoordinator is responsible for creating and uploading new media
+/// items, independently of a specific view controller. It should be accessed
+/// via the `shared` singleton.
```
[the remark this answers is not in the document]
Comment (frosty) [reply WordPress/Classes/Services/MediaUploadCoordinator.swift]: Good idea

**Comment by frosty** [comment]
@SergioEstevao Ready for another look!

**Pull request merged**